59.md

```diff
@@ -0,0 +1,250 @@
+NIP-59
+======
+
+Gift Wrap
+---------
+
+`optional`
+
+This NIP defines a protocol for encapsulating any nostr event. This makes it possible to obscure most metadata
+for a given event, perform collaborative signing, and more.
+
+This NIP relies on [NIP-44](./44.md)'s versioned encryption algorithms.
+
+# Overview
+
+This protocol uses three main concepts to protect the transmission of a target event: `rumor`s, `seal`s, and `gift wrap`s.
+
+- A `rumor` is a regular nostr event, but is **not signed**. This means that if it is leaked, it cannot be verified.
+- A `rumor` is serialized to JSON, encrypted, and placed in the `content` field of a `seal`. The `seal` is then
+  signed by the author of the note. The only information publicly available on a `seal` is who signed it, but not what was said.
+- A `seal` is serialized to JSON, encrypted, and placed in the `content` field of a `gift wrap`.
+
+This allows the isolation of concerns across layers:
+
+- A rumor carries the content but is unsigned, which means if leaked it will be rejected by relays and clients,
+  and can't be authenticated. This provides a measure of deniability.
+- A seal identifies the author without revealing the content or the recipient.
+- A gift wrap can add metadata (recipient, tags, a different author) without revealing the true author.
+
+# Protocol Description
+
+## 1. The Rumor Event Kind
+
+A `rumor` is the same thing as an unsigned event. Any event kind can be made a `rumor` by removing the signature.
+
+## 2. The Seal Event Kind
+
+A `seal` is a `kind:13` event that wraps a `rumor` with the sender's regular key. The `seal` is **always** encrypted
+to a receiver's pubkey but there is no `p` tag pointing to the receiver. There is no way to know who the rumor is for
+without the receiver's or the sender's private key. The only public information in this event is who is signing it.
+
+```js
+{
+  "id": "<id>",
+  "pubkey": "<real author's pubkey>",
+  "content": "<encrypted rumor>",
+  "kind": 13,
+  "created_at": 1686840217,
+  "tags": [],
+  "sig": "<real author's pubkey signature>"
+}
+```
+
+Tags MUST must always be empty in a `kind:13`. The inner event MUST always be unsigned.
+
+## 3. Gift Wrap Event Kind
+
+A `gift wrap` event is a `kind:1059` event that wraps any other event. `tags` MUST include a single `p` tag
```

Comment:
Or, since there are privacy implications, we may just need to define a separate wrap kind where tags are always empty and the pubkey is not random. 

The random pubkey kind is like UDP (just push events) and what you want is a TCP-like kind (ack-first-then-push). 

Either wrap can carry the same DM kinds. And you can define your authorization-to-receive-scheme in the wrap kind itself.

Reply:
Removed this part and added some clarifications that NIP 59 shouldn't be used on its own.